<?xml version="1.0" encoding="utf-8"?>
<rss xmlns:atom="http://www.w3.org/2005/Atom" version="2.0"><channel><title>MariaDB Knowledge Base Comments for: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/feed/</link><description></description><atom:link href="https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/feed/" rel="self"></atom:link><language>en-us</language><lastBuildDate>Tue, 27 Feb 2024 03:43:42 +0000</lastBuildDate><item><title>Re: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5945</link><description>&lt;p&gt;I am experiencing the same issue when attempting to connect in a shell/notebook environment (macOS Monterey 12.5.1).&lt;/p&gt;
&lt;p&gt;However, my application can connect in a container using Python 3.9 base image--which uses Debian 11?
&lt;a href="https://pythonspeed.com/articles/base-image-python-docker-images/"&gt;https://pythonspeed.com/articles/base-image-python-docker-images/&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;So this is an OS based issue?&lt;/p&gt;
</description><dc:creator xmlns:dc="http://purl.org/dc/elements/1.1/">Ben Sandham (US)</dc:creator><guid>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5945</guid></item><item><title>Re: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5934</link><description>&lt;p&gt;I think someone from mariadb needs to make the request. The issue was closed with a link to how to update the keg.&lt;/p&gt;
</description><dc:creator xmlns:dc="http://purl.org/dc/elements/1.1/">Maximilian Ebert</dc:creator><guid>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5934</guid></item><item><title>Re: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5933</link><description>&lt;p&gt;All fixed here -&amp;gt; &lt;a href="https://github.com/mariadb-corporation/mariadb-connector-c/commit/4f715e8fb1757a9c3b432471dbbc36787da6030c"&gt;https://github.com/mariadb-corporation/mariadb-connector-c/commit/4f715e8fb1757a9c3b432471dbbc36787da6030c&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;Reported to brew to update the keg (is that the right term?) -&amp;gt; &lt;a href="https://github.com/Homebrew/brew/issues/13846"&gt;https://github.com/Homebrew/brew/issues/13846&lt;/a&gt;&lt;/p&gt;
</description><dc:creator xmlns:dc="http://purl.org/dc/elements/1.1/">Maximilian Ebert</dc:creator><guid>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5933</guid></item><item><title>Re: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5914</link><description>&lt;p&gt;Just to add that I'm seeing the same thing on 2 Macs:&lt;/p&gt;
&lt;p&gt;sqlalchemy.exc.OperationalError: (mariadb.OperationalError) Can't connect to server on '192.168.50.214' (60)&lt;/p&gt;
&lt;p&gt;This is after brew upgrade MariaDB-connector-c to 3.3.2 and now there's no going back.&lt;/p&gt;
</description><dc:creator xmlns:dc="http://purl.org/dc/elements/1.1/">Andrew Collins</dc:creator><guid>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5914</guid></item><item><title>Re: mariadb-connector-c 3.3.2 on macOS doesn't connect to remote MariaDB sever</title><link>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5907</link><description>&lt;p&gt;When I looked on the SO yesterday I couldn't come to much of a conclusion.&lt;/p&gt;
&lt;p&gt;Looking at the &lt;a href="https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/intro.2.html#//apple_ref/doc/man/2/intro"&gt;OSX error&lt;/a&gt; for 60 corresponds to ETIMEDOUT.&lt;/p&gt;
&lt;p&gt;Do both servers have a equal access to the remote server they are connecting to?&lt;/p&gt;
&lt;p&gt;I looked though the &lt;a href="https://github.com/mariadb-corporation/mariadb-connector-c/compare/v3.3.1..v3.3.2"&gt;code differences&lt;/a&gt; and nothing particularly struck me as connection related.&lt;/p&gt;
&lt;p&gt;Can you &lt;a href="https://www.unix.com/man-page/osx/1/truss/"&gt;truss&lt;/a&gt; the python program on both servers to maybe compare the system calls and put then in a &lt;a href="https://jira.mariadb.org/"&gt;bug report&lt;/a&gt; (CONC project).&lt;/p&gt;
&lt;p&gt;If you see something at the OS layer returning the 60/ETIMEDOUT use a debugger, set the system call as a breakpoint, and when the error is returned, gain a full backtrace.&lt;/p&gt;
</description><dc:creator xmlns:dc="http://purl.org/dc/elements/1.1/">Daniel Black</dc:creator><guid>https://mariadb.com/kb/en/mariadb-connector-c-332-on-macos-doesnt-connect-to-remote-mariadb-sever/+comments/5907</guid></item></channel></rss>